server/src/services/InitGoogle.ts

```diff
@@ -159,14 +163,35 @@ async function createCredentialedClient() {
 // It's not necessary to recreate the client for each request, but it is safer.
 // If multiple tokens end up being used, it will be important to use the
 // appropriate one for that request.
-export async function createCalendarApi() {
-  const auth = await createCredentialedClient();
-  return calendar({ version: 'v3', auth });
+export async function createCalendarApi(
+  createClient: () => Promise<OAuth2Client> = async () =>
+    createCredentialedClient(
+      await prisma.google_tokens.findFirstOrThrow({
+        where: { is_valid: true },
+      }),
+    ),
+) {
+  return calendar({ version: 'v3', auth: await createClient() });
+}
+
+export async function createCalendarApis() {
+  const tokensInfo = await prisma.google_tokens.findMany({
+    where: { is_valid: true },
+  });
+  const apis = [];
+  for (const tokenInfo of tokensInfo) {
+    apis.push({
+      tokenId: tokenInfo.id,
+      createApi: () =>
+        createCalendarApi(() => createCredentialedClient(tokenInfo)),
+    });
+  }
+  return apis;
```

Comment:
```suggestion
  return await Promise.all(
    tokensInfo.map(async (tokenInfo) => ({
      tokenId: tokenInfo.id,
      calendarApi: await createCalendarApi(tokenInfo),
    })),
  );
```
I prefer this approach for a few reasons. It's functional, it flattens the Promises (it's a Promise, not a Promise of Promises) and finally you don't have to name as many things :D

Reply:
:D I need to do this more often, because trying to find various names is not coming that great.